src/Commands/ElasticsearchIndexer.php

```diff
@@ -34,19 +36,23 @@ public function index(iterable|object $data, callable|array|null $dataFilter, ca
         }
     }
 
-    public function indexItems(iterable $items, callable|array|null $dataFilter, callable|string $id = 'entity_id'): void
+    public function indexItems(iterable $items, callable|array|null $dataFilter = null, callable|string $id): void
     {
         foreach ($items as $item) {
             $this->indexItem($item, $dataFilter, $id);
         }
     }
 
-    public function indexItem(object $item, callable|array|null $dataFilter, callable|string $id = 'entity_id'): void
+    public function indexItem(mixed $item, callable|array|null $dataFilter = null, callable|string $id): void
     {
         if (is_null($item)) {
             return;
         }
 
+        if (! $this->prepared) {
+            throw new Exception('Attempted to index items without preparing the indexer first.');
```

Comment:
Direct exceptions are difficult to catch in an isolated way, maybe there is a more appropriate exception for this in the Elasticsearch packages